catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/builder/JsonModelBuilder.java

```diff
@@ -102,129 +82,271 @@ public FilterNode getResult() {
+  public JsonModelBuilder isEqualTo(boolean matchCase) {
+    beginTerminalType("=");
     return this;
   }
 
   @Override
-  public FlatFilterBuilder beginBinaryTemporalType(String operator) {
-    verifyResultNotYetRetrieved();
-    verifyTerminalNodeNotInProgress();
-    String jsonOperator = BINARY_TEMPORAL_MAPPING.get(operator);
-    if (jsonOperator == null) {
-      throw new IllegalArgumentException(
-          "Cannot find mapping for binary temporal operator: " + operator);
-    }
-    nodeInProgress = new FilterNodeImpl(jsonOperator);
+  public JsonModelBuilder isNotEqualTo(boolean matchCase) {
+    beginTerminalType("!=");
     return this;
   }
 
   @Override
-  public JsonModelBuilder beginBinarySpatialType(String operator) {
+  public JsonModelBuilder isGreaterThan(boolean matchCase) {
+    beginTerminalType(">");
+    return this;
+  }
+
+  @Override
+  public JsonModelBuilder isGreaterThanOrEqualTo(boolean matchCase) {
+    beginTerminalType(">=");
+    return this;
+  }
+
+  @Override
+  public JsonModelBuilder isLessThan(boolean matchCase) {
+    beginTerminalType("<");
+    return this;
+  }
+
+  @Override
+  public JsonModelBuilder isLessThanOrEqualTo(boolean matchCase) {
+    beginTerminalType("<=");
+    return this;
+  }
+
+  @Override
+  public JsonModelBuilder like(
+      boolean matchCase, String wildcard, String singleChar, String escape) {
+    String operator = (matchCase) ? "LIKE" : "ILIKE";
+    beginTerminalType(operator);
+    return this;
+  }
+
+  @Override
+  public JsonModelBuilder before() {
+    beginTerminalType("BEFORE");
+    return this;
+  }
+
+  @Override
+  public JsonModelBuilder after() {
+    beginTerminalType("AFTER");
+    return this;
+  }
+
+  @Override
+  public JsonModelBuilder intersects() {
+    beginTerminalType("INTERSECTS");
+    return this;
+  }
+
+  @Override
+  public JsonModelBuilder dwithin(double distance, String units) {
+    throw new UnsupportedOperationException("Distance buffers in JSON are not yet supported");
+  }
+
+  @Override
+  public JsonModelBuilder function(String name) {
     verifyResultNotYetRetrieved();
-    verifyTerminalNodeNotInProgress();
-    if (!BINARY_SPATIAL_OPS.contains(operator)) {
-      throw new IllegalArgumentException("Invalid operator for binary spatial type: " + operator);
-    }
-    nodeInProgress = new FilterNodeImpl(operator);
+    verifyTerminalNodeInProgress(); // TODO - Verify this; per schema functions can be predicates
+    supplierInProgress =
+        new UnboundedNodeSupplier<>(maps -> reduceFunction(name, maps), supplierInProgress);
     return this;
   }
 
   @Override
-  public JsonModelBuilder endTerminalType() {
+  public JsonModelBuilder property(String name) {
+    setProperty(name);
+    return this;
+  }
+
+  @Override
+  public JsonModelBuilder value(String value) {
+    setValue(value);
+    return this;
+  }
+
+  @Override
+  public JsonModelBuilder value(boolean value) {
+    setValue(Boolean.toString(value));
```

Comment:
❗️ Address TODO. This might have been a side effect of JSON parsing. Revisit correct re-hydration of serializable objects into primitive types. 

_See companion comment in XmlModelBuilder as well_.

Reply:
Fixed.